File: gestor-de-tareas-en-go/tarea.go
package main

import (
	"encoding/json"
	"fmt"
	"time"
)

type EstadoTarea int

const (
	Pendiente EstadoTarea = iota
	EnProgreso
	Completada
)

func (et EstadoTarea) String() string {
	switch et {
	case Pendiente:
		return "Pendiente"
	case EnProgreso:
		return "En Progreso"
	case Completada:
		return "Completada"
	default:
		return "Desconocido"
	}
}

func (et EstadoTarea) MarshalJSON() ([]byte, error) {
	return json.Marshal(et.String())
}

func (et *EstadoTarea) UnmarshalJSON(b []byte) error {
	var s string
	if err := json.Unmarshal(b, &s); err != nil {
		return err
	}
	switch s {
	case "Pendiente":
		*et = Pendiente
	case "En Progreso":
		*et = EnProgreso
	case "Completada":
		*et = Completada
	default:
		return fmt.Errorf("estado de tarea desconocido: %s", s)
	}
	return nil
}

type Tarea struct {
	ID            int         `json:"id"`
	Titulo        string      `json:"titulo"`
	Estado        EstadoTarea `json:"estado"`
	FechaCreacion time.Time   `json:"fecha_creacion"`
	FechaEdicion  *time.Time  `json:"fecha_edicion,omitempty"`
}

type Tareas []Tarea

File: gestor_tareas_rust/src/tarea.rs
use serde::{Serialize, Deserialize};
use chrono::{DateTime, Local};
use std::fmt;

#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
pub enum Estado {
    Pendiente,
    EnProgreso,
    Completada,
}

impl fmt::Display for Estado {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        match self {
            Estado::Pendiente => write!(f, "Pendiente"),
            Estado::EnProgreso => write!(f, "En Progreso"),
            Estado::Completada => write!(f, "Completada"),
        }
    }
}

#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
pub struct Tarea {
    pub id: u32,
    pub titulo: String,
    pub descripcion: String,
    pub estado: Estado,
    pub fecha_creacion: DateTime<Local>,
    pub fecha_edicion: Option<DateTime<Local>>,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct GestorTareas {
    pub tareas: Vec<Tarea>,
    pub siguiente_id: u32,
}